src/main/java/service/FruitShopService.java

```diff
@@ -0,0 +1,36 @@
+package service;
+
+import exception.OperationException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import model.OperationType;
+import strategy.OperationHandler;
+
+public class FruitShopService {
+    private final Map<String, Integer> fruitInventory = new HashMap<>();
+    private final Map<OperationType, OperationHandler> operationStrategy;
+
+    public FruitShopService(Map<OperationType, OperationHandler> operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    public void processTransactions(List<FruitTransaction> transactions) {
```

Comment:
Ensure that the `processTransactions` method doesn't violate the encapsulation principle by modifying the `fruitInventory` directly. Instead, the operation handlers should be responsible for any changes to the inventory.